### ISSUE TITLE
Display `${media-title}` in playlist instead of `${filename}`?

### ISSUE THREAD

**Issue by xfzv** [tomasklaen/uosc#422]
I know it's not implemented for online media (https://github.com/tomasklaen/uosc/issues/395) but is it supposed to work with local files?

![1](https://user-images.githubusercontent.com/78810647/211517032-ceeefe0e-68c9-4ecc-8b19-594dd5765dc5.png)

If not, any chance it could be added? something like:

```
playlist_entry=${media-title}
playlist_entry=${filename}
```

**Comment by tomasklaen** [comment]
This is not available. All you get is these values: https://mpv.io/manual/master/#command-interface-playlist

Meaning `title` and `filename`, and we are already using `title` and fall back to `filename` when it's not available. Any template expansion is not possible here, nor practical.

Just think about it. mpv would have to read meta data for every file in the playlist, which would be very slow for huge playlists.

In other words, we are already displaying the most fitting title that mpv provides.

**Comment by tomasklaen** [comment]
> 10 seconds ... it isn't slow at all

I guess we have a very different definition of _slow_ then :)

The only fast and practical solution here would be to create a cross-platform meta data server that we could query for file meta of only the items currently visible in the UI (not the ones currently culled away). But that would require a lot of development time, and probably additional installation steps.

Even if I'd like to see better titles in playlists myself, the degree to which this bothers me is orders of magnitude lower than the effort that'd be required to improve it, so I don't see myself working on it.

Though if mpv exposed its file/url meta retrieval as an API function, that would make it easy. But even then, mpv gets meta for urls from youtube-dl, which is still really slow, so we'd have to roll our own url title resolver that would be getting them from the `<title>` tag of its html response. So we'd _still_ need a server.
